### PR TITLE
Relocate and consolidate tests

### PR DESCRIPTION
**Pull request by nickrnet**
Get a common theme for ongoing project maintenance.

Gives us a listing of:
```
.
├── README.md
├── api.properties
├── doc
│   ├── development
│   │   └── README.md
│   └── screenshots
│       ├── dashboard.png
│       ├── players.png
│       ├── server.png
│       └── world.png
├── index.js
├── minecraft-server.log
├── minecraft_server
│   ├── 1.11.2_minecraft_server.jar
│   ├── 1.14.4_minecraft_server.jar
│   ├── backups
│   │   └── worlds
│   ├── banned-ips.json
│   ├── banned-players.json
│   ├── eula.txt
│   ├── logs
│   │   └── latest.log
│   ├── ops.json
│   ├── server.jar
│   ├── server.properties
│   ├── usercache.json
│   ├── whitelist.json
│   └── world
├── node_modules
├── package-lock.json
├── package.json
├── release-notes.txt
├── server.properties
├── src
│   ├── api
│   │   └── MinecraftApi.js
│   ├── server
│   │   ├── Eula.js
│   │   └── MinecraftServer.js
│   ├── util
│   │   └── util.js
│   └── web
│       ├── build
│       ├── node_modules
│       ├── package-lock.json
│       ├── package.json
│       ├── public
│       └── src
│           ├── About
│           ├── App.css
│           ├── App.js
│           ├── App.test.js
│           ├── Dashboard
│           ├── Players
│           ├── Preferences
│           ├── ServerControls
│           ├── index.css
│           ├── index.js
│           ├── logo.svg
│           └── registerServiceWorker.js
└── test
    ├── api
    │   └── MinecraftApi.js
    ├── results
    └── server
        ├── Eula.js
        └── MinecraftServer.js
```
so that as we expand the backend classes (breaking apart the MinecraftServer and API separately), it stays sane with a testing structure that matches the src structure.

I'm still debating moving the web UI out into its own repo, but for now it stays until some kind of better build process.

### CODE REVIEW

**Comment by nickrnet** [comment]
@DevBonBon thoughts?

**Comment by fluvf** [comment]
Sure! I have a preference on how the folder structure should look like, but that's all it is, a preference
I'll try out the tests and give a quick review

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "start": "node index.js",
     "start-dev": "pushd minecraftservermanager && ./node_modules/.bin/react-scripts start && popd",
     "build-web": "pushd minecraftservermanager && ./node_modules/.bin/react-scripts build && popd",
-    "test-backend": "./node_modules/mocha/bin/_mocha -u bdd --colors --reporter mochawesome --reporter-options reportDir=./src/tests/results src/test",
+    "test-backend": "./node_modules/mocha/bin/_mocha -u bdd --colors --reporter mochawesome --reporter-options reportDir=./test/results src/test",
```
Comment (fluvf) [inline package.json]: `mocha --recursive --reporter mochawesome --reporter-options reportDir=test/results`
- Using `./---` relative paths breaks on windows apparently, just `mocha` should be fine?
- `-u bdd -colors` are the defaults, so removing them doesn't hurt
- There's nothing in `src/test` anymore
- `--recursive` as tests are in sub-folders now

**Review by fluvf — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nickrnet) [reply package.json]: fixed. I don't see a `src/test` in my checkout, if it's still around after merge, I'll just `git rm` it on master directly 😈

**Review by fluvf — APPROVED** [review]

**Pull request merged**